### PR TITLE
DDF-4055 Changed method to be final and fixed unit tests

### PR DESCRIPTION
**Pull request by tyler30clemens**
##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Link to 2.13.x PR: #3584 
____

#### What does this PR do?
Prevents overriding of the Security.getSystemSubject method to prevent malicious persons from replacing secured versions with unsecured versions of classes/services.
forward port of #3584

#### Who is reviewing it? 
@jhunzik @ethantmanns 

#### Ask 2 committers to review/merge the PR and tag them here.
@tbatie @figliold 

#### How should this be tested?

1. Get changes (make sure you have changes from CAL-460 or else some tests in CAL will break due to not being able to mock final methods) and test respective unit tests.
2. Install then run DDF and hit Security.getSystemSubject method and make sure nothing breaks.

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by tyler30clemens** [comment]
build now

**Comment by bakejeyner** [comment]
Hero ✔️

File: catalog/spatial/registry/registry-federation-admin-service-impl/pom.xml [hunk 1]
```diff
@@ -95,6 +95,13 @@
             <artifactId>filter-proxy</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <!-- Upgrade mockito to version 2.1.0 for this module to get final mock capability -->
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>2.1.0</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (ahoffer) [inline catalog/spatial/registry/registry-federation-admin-service-impl/pom.xml]: @lessarderic We should probably have a strategy for overriding the managed version of a library. What are your thoughts?

**Comment by tyler30clemens** [comment]
build now

**Comment by tyler30clemens** [comment]
build now

**Comment by tyler30clemens** [comment]
build now 
(flaky install uninstall)

**Comment by tyler30clemens** [comment]
build now

**Pull request merged**